iroh-bytes/src/provider/database.rs

```diff
@@ -31,6 +36,156 @@ pub const FNAME_PATHS: &str = "paths.bin";
 #[derive(Debug, Clone, Default)]
 pub struct Database(Arc<RwLock<HashMap<Hash, DbEntry>>>);
 
+#[derive(Debug, Clone, Default)]
+pub struct InMemDatabase(Arc<HashMap<Hash, (PreOrderMemOutboard, Bytes)>>);
```

Comment:
maybe make a `database` mod, these files are getting pretty big

Reply:
If I had done this I would have been in even more of a merge hell than I was already... So maybe leave it for the next PR?

Reply:
sure